### ISSUE TITLE
[Bug]: controlnet not working 

### ISSUE THREAD

**Issue by alexfierson** [Mikubill/sd-webui-controlnet#1453]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

unable to use controlnet

### Steps to reproduce the problem

so i open sd and start making waifu and stuff, but notice that controlnet was disabled even if i check and unchecked the enable  box, and the control mode & resize mode nodes also unchecked themself after each generated picture

### What should have happened?

it should work, like how i used it 8 hours ago

### Commit where the problem happens

webui 1.3.0 
python 3.10.11 
gardio 3.31.0 
cozy nest ui 2.1.6

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
-- xformer -- medvram
```

### List of enabled extensions

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/134821380/b913886c-a8c4-4b07-84ec-a73c9292072f)

### Console logs

```Shell
Error running process: C:\Users\HKC\Desktop\WD40\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py

Traceback (most recent call last):
  File "C:\Users\HKC\Desktop\WD40\stable-diffusion-webui\modules\scripts.py", line 451, in process
    script.process(p, *script_args)
  File "C:\Users\HKC\Desktop\WD40\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1035, in process
    self.enabled_units = self.get_enabled_units(p)
  File "C:\Users\HKC\Desktop\WD40\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 994, in get_enabled_units
    unit = self.parse_remote_call(p, unit, idx)
  File "C:\Users\HKC\Desktop\WD40\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 844, in parse_remote_call
    unit.enabled = selector(p, "control_net_enabled", unit.enabled, idx, strict=True)
AttributeError: 'list' object has no attribute 'enabled'
```

### Additional information

i don't know much, but all things work fine at gradio 3.29

**Comment by eyeweaver** [comment]
I have the same issue

**Comment by Joeyd348039** [comment]
I just started getting the issue out of nowhere. Controlnet was working fine one moment and then it just stopped. I updated, restarted and no luck.

I did notice that the Preprocessor Preview no longer works. 
It also shows this message:
"Calling preprocessor reference_only outside of cache."
And right below that there is no loading, so it stays at 0%. 

So I hope this might be a clue as to what is going on.

**Comment by lllyasviel** [comment]
extension conflict. please disable all other extensions and test them one by one.

**Comment by Joeyd348039** [comment]
> 

Tried that still no luck. I'm not sure if this is an error but it now says this:

Create LRU cache (max_size=16) for preprocessor results.

**Comment by lllyasviel** [comment]
update CN and webui, disable all other extensions, completely restart terminal
this can solve all problems.
then try to enable other extensions and see which caused problem

**Comment by Joeyd348039** [comment]
> update CN and webui, disable all other extensions, completely restart terminal this can solve all problems. then try to enable other extensions and see which caused problem

Just went through everything.Update, disabling and restarting terminal, same issues. The only way to get rid of that error is by disabling SD-WEBUI-CONTROLNET .

**Comment by lllyasviel** [comment]
> > update CN and webui, disable all other extensions, completely restart terminal this can solve all problems. then try to enable other extensions and see which caused problem
> 
> Just went through everything.Update, disabling and restarting terminal, same issues. The only way to get rid of that error is by disabling SD-WEBUI-CONTROLNET .

can you provide more infos to reproduce the problem?

**Comment by lllyasviel** [comment]
Also if more people get here, take a look at
https://github.com/Mikubill/sd-webui-controlnet/issues/1289
and
https://github.com/Mikubill/sd-webui-controlnet/issues/1289

Most problems can be resolved by disabling webui extensions or browser extensions

**Comment by Joeyd348039** [comment]
> can you provide more infos to reproduce the problem?

Not sure what else I can provide. I also installed the new A1111 script with nothing installed, but as soon as I go to install Controlnet I get the same error.

Heres the usual error:
Loading preprocessor: reference_only█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 16/16 [00:06<00:00,  2.62it/s]
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.INNER_FIT
raw_H = 675
raw_W = 450
target_H = 512
target_W = 512
estimation = 512.0
preprocessor resolution = 512
  0%|                                                                                                                                                     | 0/16 [00:00<?, ?it/s]ControlNet used torch.float32 VAE to encode torch.Size([1, 4, 64, 64]).

---

Not sure if that helps.

**Comment by lllyasviel** [comment]
0/16 stuck here and it does not move on?

**Comment by lllyasviel** [comment]
If that is the case, this is probably because A1111 1.3.0 used new backends for attention backends.  I forget where to set but you can find some info from a1111’s new release and find out how to shift to other or old attention backends.

**Comment by Joeyd348039** [comment]
0/16 stays like that and then it just moves onto making the images. 

I tried the A1111 and VLAD's fork that both worked with Controlnet up until 1 day ago.  I also get the error when the scripts are done loading like this:
"Create LRU cache (max_size=16) for preprocessor results.
Create LRU cache (max_size=16) for preprocessor results.
Running on local URL:  http://127.0.0.1:7860 "

On the new A1111 there is nothing installed on there either and still no luck.

**Comment by lllyasviel** [comment]
these are not errors.
if it can generate images then it is working as expected. I am going to close this.
Feel free to reopen if new problems are discovered.

**Comment by DenisSergeevitch** [comment]
Have the same error, traceback; WebUI part also does not load well:
```
Error loading script: controlnet.py
Traceback (most recent call last):
  File "E:\notebooks\stable-diffusion-webui\modules\scripts.py", line 263, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "E:\notebooks\stable-diffusion-webui\modules\script_loading.py", line 10, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 14, in <module>
    importlib.reload(global_state)
  File "C:\Users\i\AppData\Local\Programs\Python\Python310\lib\importlib\__init__.py", line 169, in reload
    _bootstrap._exec(spec, module)
  File "<frozen importlib._bootstrap>", line 619, in _exec
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\global_state.py", line 9, in <module>
    from scripts.utils import ndarray_lru_cache
ImportError: cannot import name 'ndarray_lru_cache' from 'scripts.utils' (E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\utils.py)

ControlNet v1.1.204
Error loading script: global_state.py
Traceback (most recent call last):
  File "E:\notebooks\stable-diffusion-webui\modules\scripts.py", line 263, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "E:\notebooks\stable-diffusion-webui\modules\script_loading.py", line 10, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\global_state.py", line 9, in <module>
    from scripts.utils import ndarray_lru_cache
ImportError: cannot import name 'ndarray_lru_cache' from 'scripts.utils' (E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\utils.py)
```

```
Error executing callback app_started_callback for E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\api.py
Traceback (most recent call last):
  File "E:\notebooks\stable-diffusion-webui\modules\script_callbacks.py", line 125, in app_started_callback
    c.callback(demo, app)
  File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\api.py", line 54, in controlnet_api
    cached_cn_preprocessors = global_state.cache_preprocessors(global_state.cn_preprocessor_modules)
AttributeError: module 'scripts.global_state' has no attribute 'cache_preprocessors'
```

WebUI version: version: v1.3.0

**Comment by huchenlei** [comment]
> Have the same error, traceback; WebUI part also does not load well:
> 
> ```
> Error loading script: controlnet.py
> Traceback (most recent call last):
>   File "E:\notebooks\stable-diffusion-webui\modules\scripts.py", line 263, in load_scripts
>     script_module = script_loading.load_module(scriptfile.path)
>   File "E:\notebooks\stable-diffusion-webui\modules\script_loading.py", line 10, in load_module
>     module_spec.loader.exec_module(module)
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 14, in <module>
>     importlib.reload(global_state)
>   File "C:\Users\i\AppData\Local\Programs\Python\Python310\lib\importlib\__init__.py", line 169, in reload
>     _bootstrap._exec(spec, module)
>   File "<frozen importlib._bootstrap>", line 619, in _exec
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\global_state.py", line 9, in <module>
>     from scripts.utils import ndarray_lru_cache
> ImportError: cannot import name 'ndarray_lru_cache' from 'scripts.utils' (E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\utils.py)
> 
> ControlNet v1.1.204
> Error loading script: global_state.py
> Traceback (most recent call last):
>   File "E:\notebooks\stable-diffusion-webui\modules\scripts.py", line 263, in load_scripts
>     script_module = script_loading.load_module(scriptfile.path)
>   File "E:\notebooks\stable-diffusion-webui\modules\script_loading.py", line 10, in load_module
>     module_spec.loader.exec_module(module)
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\global_state.py", line 9, in <module>
>     from scripts.utils import ndarray_lru_cache
> ImportError: cannot import name 'ndarray_lru_cache' from 'scripts.utils' (E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\utils.py)
> ```
> 
> ```
> Error executing callback app_started_callback for E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\api.py
> Traceback (most recent call last):
>   File "E:\notebooks\stable-diffusion-webui\modules\script_callbacks.py", line 125, in app_started_callback
>     c.callback(demo, app)
>   File "E:\notebooks\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\api.py", line 54, in controlnet_api
>     cached_cn_preprocessors = global_state.cache_preprocessors(global_state.cn_preprocessor_modules)
> AttributeError: module 'scripts.global_state' has no attribute 'cache_preprocessors'
> ```
> 
> WebUI version: version: v1.3.0

Close WebUI terminal and restart should solve your problem. It is because A1111 not picking changes we made in some scripts due to caching.

**Comment by DenisSergeevitch** [comment]
It helped. Thank you!

**Comment by Joeyd348039** [comment]
> these are not errors. if it can generate images then it is working as expected. I am going to close this. Feel free to reopen if new problems are discovered.

It isn't generating images based on reference. It makes no changes. I installed other models and they work 100%. The PreProcessor goes to 100% and they work properly. It's almost like the script is  looking for the reference_only model and isn't finding it.

**Comment by lokobb** [comment]
Hello everyone, I'm not a programmer or a computer expert, and I encountered the same issue while researching how to use it. I couldn't find the answer online, but I managed to solve the problem. I know how frustrating it can be, so I just created a GitHub account to reply and hopefully provide some help! 

Actually, the solution is very simple. Take a look at your 'control_sd15_openpose.pth' file. It should only be a few tens of kilobytes, right? The reason is that you downloaded it using the 'right-click save as' option. Go to the download page an just click 'download' to download the 5GB file. If that doesn't work, try reinstalling it again!
